tests/examples/examples_ft.cpp

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <iostream> // TODO remove
+#include <string>
+#include <regex>
+#include <fstream>
+#include <vector>
+#include <k4a/k4a.h>
+#include <utcommon.h>
+#include <gtest/gtest.h>
+
+const static std::string TEST_TEMP_DIR = "examples_test_temp";
+
+// run the specified executable and record the output to output_path
+// if output_path is empty, just output to stdout
+static void run_and_record_executable(const std::string &executable_path,
+                                      const std::string &args,
+                                      const std::string &output_path)
+{
+    // TODO fflush needed here?
+    std::string formatted_command = executable_path;
+    formatted_command += " " + args;
+    if (!output_path.empty())
+    {
+        formatted_command += " 2>&1 > " + output_path;
+    }
+    std::cout << formatted_command << std::endl;
+#ifdef _WIN32
+    FILE *process_stream = _popen(formatted_command.c_str(), "r");
+#else
+    FILE *process_stream = popen(formatted_command.c_str(), "r");
+#endif
+    ASSERT_TRUE(process_stream);
+
+    // TODO errno?
+    // ASSERT_NE(fgets(buffer, 10000, first), nullptr);
+    // TODO temporary, make better
+    // Not in hereASSERT_NE(process_stream, nullptr);, EXIT_SUCCESS
+#ifdef _WIN32
+    ASSERT_EQ(_pclose(process_stream), EXIT_SUCCESS);
+#else
+    ASSERT_EQ(pclose(process_stream), EXIT_SUCCESS);
+#endif
+}
+
+static void test_stream_against_regexes(std::istream &input_stream, const std::vector<std::string> &regexes)
+{
+    auto regex_iter = regexes.cbegin();
+    // ensure that all regexes are matched before the end of the file, in this order
+    while (input_stream && regex_iter != regexes.cend())
+    {
+        std::string results;
+        getline(input_stream, results);
+        std::regex desired_out(*regex_iter);
+        std::cout << results << std::endl; // TODO delete this
+        if (std::regex_match(results, desired_out))
+        {
+            ++regex_iter;
+        }
+    }
+    ASSERT_EQ(regex_iter, regexes.cend()) << "Reached the end of the example output before matching all of the "
+                                             "required regular expressions.\nExpected \""
+                                          << *regex_iter << "\", but never saw it.";
+}
+
+class examples_ft : public ::testing::Test
+{
+protected:
+    void SetUp() override
+    {
+#ifdef _WIN32
+        // _putenv_s("K4A_ENABLE_LOG_TO_STDOUT", "0");
```

Comment:
Why is this commented out? If there's a reason this should behave differently on different platforms, a comment would be good here.